pkg/config/config.go

```diff
@@ -201,22 +199,11 @@ func normalizeConfigLoadPath(target string) (string, error) {
 	return configPath, nil
 }
 
-// tryLoadConfig tries to load config in `target` (or it's containing directory)
-// `target` will be the key for the entry in configMap
+// tryLoadConfig attempts to parse the config file at the given path as TOML,
+// returning the Config object if successful or otherwise the error
 func tryLoadConfig(configPath string) (Config, error) {
-	file, err := os.Open(configPath)
-	var config Config
-	if err == nil { // File exists, and we have permission to read
-		defer file.Close()
-
-		_, err := toml.NewDecoder(file).Decode(&config)
-		if err != nil {
-			return Config{}, fmt.Errorf("failed to parse config file: %w", err)
-		}
-		config.LoadPath = configPath
-
-		return config, nil
-	}
+	config := Config{LoadPath: configPath}
+	_, err := toml.DecodeFile(configPath, &config)
```

Comment:
note: I've realised this actually makes it possible to "configure" the `LoadPath`

I will look back into this shortly

Reply:
I ended up just refactoring to only set the `LoadPath` if `err` is `nil` - while it makes the function a bit bigger, it means we maintain the existing public behaviour without costing us a lot, and (importantly) lets us defer a bikeshed 😅

I did consider that it might be useful to have the `LoadPath` set in all cases since in the event of an error you could use that to programmatically know what file it was erroring on, but again that'd mean doing some bikeshedding and frankly I don't think it's useful enough to do it for now when we can easily change the behaviour later 🤷